src/ui-components/share-dialog.tsx

```diff
@@ -108,52 +68,66 @@ export class ShareDialog extends ReactWidget {
   }
 }
 
-// TODO: not used until the shareable link works
+/**
+ * Success dialog - shows actual URLs and passwords, minimal styling
+ * Shows password only when isNewShare is true.
+ */
 export const createSuccessDialog = (
   shareableLink: string,
   isNewShare: boolean,
-  isViewOnly: boolean
+  isViewOnly: boolean,
+  password?: string
 ) => {
   return (
     <div>
-      <p style={{ fontSize: '1.2em', marginBottom: '15px' }}>
-        {isNewShare ? 'Your notebook is now shared!' : 'Your notebook has been updated!'} Use this
-        link to access it:
-      </p>
+      <h3>
+        {isNewShare
+          ? 'Here is the shareable link to your new copy:'
+          : 'Here is the shareable link to your notebook:'}
+      </h3>
 
       <div
         style={{
-          textAlign: 'center',
-          margin: '15px 0',
+          backgroundColor: '#f0f0f0',
           padding: '10px',
-          background: '#f5f5f5',
-          borderRadius: '4px'
+          borderRadius: '5px',
+          marginBottom: '20px',
+          wordBreak: 'break-all',
+          fontFamily: 'monospace'
         }}
       >
-        <a
-          href={shareableLink}
-          target="_blank"
-          rel="noopener noreferrer"
-          style={{
-            fontSize: '1.1em',
-            color: '#007bff',
-            textDecoration: 'underline',
-            wordBreak: 'break-all'
-          }}
-        >
-          {shareableLink}
-        </a>
+        {shareableLink}
       </div>
 
-      {isViewOnly && (
-        <p style={{ marginTop: '15px' }}>
-          <strong>Note:</strong> This notebook is password-protected.
-        </p>
+      {isNewShare && password && (
+        <>
+          <p>
+            Here's the code required to edit the original notebook. Make sure to save this code as
+            it will not appear again:
+          </p>
+          <div
+            style={{
+              backgroundColor: '#f0f0f0',
+              padding: '10px',
+              borderRadius: '5px',
+              marginBottom: '20px',
+              fontFamily: 'monospace',
+              fontSize: '14px',
+              letterSpacing: '1px'
+            }}
+          >
+            {password}
+          </div>
+        </>
       )}
     </div>
   );
 };
 
+/**
+ * An error dialog for sharing failures.
+ * Displays a generic error message.
+ */
```

Comment:
Thanks for the docstring. I'd also recommend using a docstring generator to keep style and fields consistent across all functions - for example, this is missing the `error` parameter if we're following the docblocks style docstrings. I use `neogen`, but there are several out there for all IDEs that can do this.

Reply:
I was able to get VS Code to generate the boilerplate for a JSDoc comment, and added it in a4c3a88da8285fe81f6f1db278d23c6d7512db20. Thanks!